R/BackendMzR-class.R

```diff
@@ -4,17 +4,17 @@ NULL
 setClass("BackendMzR", contains = "Backend")
 
 #' @rdname hidden_aliases
-setMethod("backendReadSpectra", "BackendMzR", function(object, file,
+setMethod("backendReadSpectra", "BackendMzR", function(object,
                                                        spectraData, ...,
-                                                       BPPARAM=bpparam()) {
-    spd_list <- split.data.frame(spectraData, f = spectraData$fileIdx)
-    if (length(spd_list) != length(file))
-        stop("Number of files in 'spectraData' has to match length of 'file'")
-    res <- bpmapply(FUN = .spectra_from_file_mzR, file, spd_list,
-                    BPPARAM = BPPARAM, SIMPLIFY = FALSE)
-    names(res) <- NULL
-    res <- unlist(res, recursive = FALSE)
-    res[rownames(spectraData)]
+                                                       BPPARAM = bpparam()) {
+    file_f <- factor(spectraData$fileIdx, levels = unique(spectraData$fileIdx))
+    fls <- object@files[as.integer(levels(file_f))]
```

Comment:
Wouldn't be `fls <- object@files[spectraData$fileIdx]` enough, or if you are afraid of stupid (non-numeric) `fileIdx`:
```r
cfileIdx <- as.character(fileIdx)
fls <- object@files[match(cfileIdx, unique(cfileIdx))]
```
(should be a function)

Reply:
What I wanted to make sure here is that the `spectraData` is split how it is ordered, i.e. if (by whatever reason) `spectraData` contains first spectra from the second file and then from the first, that spectra from the second file are read first. I just wanted avoid to return the results as `res[rownames(spectraData)]`.

Reply:
What would be the benefit from your solution?

Reply:
> What would be the benefit from your solution?

None. In fact it is the same as `factor` would do internally (but with more code etc.). I am sometimes confused about how `factor` work (ordered vs unsorted, index vs names etc.). With my solution I exactly see what happens. But it doesn't matter here, keep what you like more.